MyFirstSlideshow/ViewController.swift

```diff
@@ -8,18 +8,96 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+class ViewController: UIViewController, NSURLConnectionDataDelegate {
+    // Images under Creative Commons 
+    // Images attributed to (in order) https://www.flickr.com/photos/_torne/
+    // https://www.flickr.com/photos/chrisyarzab/
+    // https://www.reddit.com/user/lalien42/
+    // http://www.kapstadt.de/
+
+    let images: Array<String> = ["https://c1.staticflickr.com/6/5615/15570202337_0e64f5046e_k.jpg",
+                                 "https://c1.staticflickr.com/4/3169/2846544061_cb7c04b46f_b.jpg",
+                                  "https://i.redd.it/d8q1wkgu1awy.jpg",
+                                  "http://www.kapstadt.de/webcam.jpg"]
+
+    var i: Int = 0
+    var j: Int = 0
+
+    @IBOutlet weak var image1: UIImageView!
+    @IBOutlet weak var image2: UIImageView!
+
+    @IBOutlet var dots1: [UIView]?
+    @IBOutlet var dots2: [UIView]?
+
+    @IBAction func onButtonOneClicked(_ sender: UIButton) {
+        let currentImage = images[i % 4]
+        i += 1
+        // the last url of the array changes every ~ 15 minutes
+        let data = try! Data(contentsOf: URL(string: currentImage)!, options: .uncached)
```

Comment:
Image is downloading each and everytime it can be cached that can prevent it to download it each time